src/transport/mqtt/config/tls.rs

```diff
@@ -127,24 +128,45 @@ impl ClientAuth {
     }
 }
 
-async fn read_root_cert_store() -> Result<RootCertStore, PairingError> {
+#[cfg(feature = "webpki")]
+#[instrument]
+pub(crate) async fn read_root_cert_store() -> Result<RootCertStore, PairingError> {
+    debug!("reading root cert store from webpki");
+
+    let root_cert_store = RootCertStore {
+        roots: webpki_roots::TLS_SERVER_ROOTS.to_vec(),
+    };
+
+    Ok(root_cert_store)
+}
+
+#[cfg(not(feature = "webpki"))]
+#[instrument]
+pub(crate) async fn read_root_cert_store() -> Result<RootCertStore, PairingError> {
+    debug!("reading root cert store from native certs");
+
     tokio::task::spawn_blocking(|| {
         let mut root_cert_store = RootCertStore::empty();
 
-        let native_certs =
-            rustls_native_certs::load_native_certs().map_err(PairingError::Native)?;
+        let rustls_native_certs::CertificateResult { certs, errors, .. } =
```

Comment:
```suggestion
        let res =
```
I think restructuring the result hinders the readability